lib/spack/spack/test/build_environment.py

```diff
@@ -191,13 +190,11 @@ def test_compiler_config_modifications(
     # Check they were applied
     for name, value in expected.items():
         if value is not None:
-            assert os.environ[name] == value
+            assert os.environ[name].replace("\\", "/") == value
```

Comment:
Not sure what this test is actually doing, but will this change translate to unix?

Reply:
I think it will be ok on Unix. To avoid confusion, I changed it to only do the check if we're on Windows